awx/api/views/root.py

```diff
@@ -40,19 +40,19 @@
 class ApiRootView(APIView):
     permission_classes = (AllowAny,)
     name = _('REST API')
-    versioning_class = None
+    versioning_class = URLPathVersioning
     swagger_topic = 'Versioning'
 
     @method_decorator(ensure_csrf_cookie)
     def get(self, request, format=None):
         '''List supported API versions'''
-
-        v2 = reverse('api:api_v2_root_view', kwargs={'version': 'v2'})
+        v2 = reverse('api:api_v2_root_view', request=request, kwargs={'version': f'v2'})
```

Comment:
add `request` so that `drf_reverse()` can know the path that the user visited the api endpoint so that it can check for `controller` in  `/api/controller/v2/` or the absence of.

 Note that `serializers.py` have a pattern of calling `self.reverse()` and that function pulls `request` from the local context and passes it to the lower level `reverse`.